error in delete cluster with LB

/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]


when delete a cluster with following command (has LB created), got panic:

/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x10435a0]

goroutine 152 [running]:
sigs.k8s.io/cluster-api-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1301540, 0x2328050)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/openstack/services/networking.(*Service).DeleteLoadBalancerMember(0xc00000e688, 0xc000043760, 0xd, 0xc000102000, 0xc0006dc5a0, 0x0, 0x0)
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/openstack/services/networking/loadbalancer.go:299 +0x330
sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/openstack/machine.(*Actuator).Delete(0xc000531e30, 0x1689fe0, 0xc00009e020, 0xc0000cf040, 0xc000102000, 0x1, 0x1)
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/openstack/machine/actuator.go:231 +0x873
sigs.k8s.io/cluster-api-provider-openstack/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine.(*ReconcileMachine).Reconcile(0xc000424500, 0xc00009f9e0, 0x7, 0xc000429be0, 0x16, 0x233c140, 0x43c73d, 0xc0002e7df8, 0x1)
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/machine_controller.go:173 +0xe33
sigs.k8s.io/cluster-api-provider-openstack/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0000c6140, 0x0)
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215 +0x1cc
sigs.k8s.io/cluster-api-provider-openstack/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1()
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158 +0x36
sigs.k8s.io/cluster-api-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00029c5e0)
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x54
sigs.k8s.io/cluster-api-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00029c5e0, 0x3b9aca00, 0x0, 0x1, 0xc000048120)
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134 +0xf8
sigs.k8s.io/cluster-api-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc00029c5e0, 0x3b9aca00, 0xc000048120)
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/cluster-api-provider-openstack/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
        /go/src/sigs.k8s.io/cluster-api-provider-openstack/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x311


**What did you expect to happen:**


**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

--- a/pkg/cloud/openstack/machine/actuator.go
+++ b/pkg/cloud/openstack/machine/actuator.go
@@ -226,7 +226,7 @@ func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machi
 		return err
 	}
 	if clusterProviderSpec.ManagedAPIServerLoadBalancer {
-		err = loadbalancerService.DeleteLoadBalancerMember(clusterName, machine, clusterProviderStatus)
+		err = loadbalancerService.DeleteLoadBalancerMember(clusterName, machine, clusterProviderSpec, clusterProviderStatus)
 		if err != nil {
 			return err
 		}

--- a/pkg/cloud/openstack/services/loadbalancer/loadbalancer.go
+++ b/pkg/cloud/openstack/services/loadbalancer/loadbalancer.go
@@ -264,7 +264,7 @@ func (s *Service) ReconcileLoadBalancerMember(clusterName string, machine *clust
 	return nil
 }
 
-func (s *Service) DeleteLoadBalancerMember(clusterName string, machine *clusterv1.Machine, clusterProviderStatus *providerv1.OpenstackClusterProviderStatus) error {
+func (s *Service) DeleteLoadBalancerMember(clusterName string, machine *clusterv1.Machine, clusterProviderSpec *providerv1.OpenstackClusterProviderSpec, clusterProviderStatus *providerv1.OpenstackClusterProviderStatus) error {
 
 	if machine == nil || !util.IsControlPlaneMachine(machine) {
 		return nil
@@ -275,13 +275,19 @@ func (s *Service) DeleteLoadBalancerMember(clusterName string, machine *clusterv
 
 	lbID := clusterProviderStatus.Network.APIServerLoadBalancer.ID
 
-	for _, port := range []int{22, 6443} {
+	portList := []int{clusterProviderSpec.APIServerLoadBalancerPort}
+	portList = append(portList, clusterProviderSpec.APIServerLoadBalancerAdditionalPorts...)
+	for _, port := range portList {
 		lbPortObjectsName := fmt.Sprintf("%s-%d", loadBalancerName, port)
 		name := lbPortObjectsName + "-" + machine.Name
 
 		pool, err := checkIfPoolExists(s.loadbalancerClient, lbPortObjectsName)
 		if err != nil {
 			return err
+		}
+		if pool == nil {
+			klog.Infof("Pool %s does not exist", lbPortObjectsName)
+			continue
 		}
 
 		lbMember, err := checkIfLbMemberExists(s.loadbalancerClient, pool.ID, name)

